src/rest_vol.c

```diff
@@ -2732,14 +2477,258 @@ RV_copy_object_loc_info_callback(char *HTTP_response, void *callback_data_in, vo
+
+    if (index >= key_obj->u.array.len)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "requested link index was out of bounds");
+
+    switch (idx_params->order) {
+        case (H5_ITER_DEC):
+            if (NULL == (link_obj = key_obj->u.array.values[key_obj->u.object.len - 1 - index]))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "selected link was NULL");
+            break;
+
+        case (H5_ITER_NATIVE):
+        case (H5_ITER_INC):
+            if (NULL == (link_obj = key_obj->u.array.values[index]))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "selected link was NULL");
+            break;
+        case (H5_ITER_N):
+        case (H5_ITER_UNKNOWN):
+        default: {
+            FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "invalid iteration order");
+            break;
+        }
+    }
+
+    /* Iterate through key/value pairs in link response to find name */
+    for (size_t i = 0; i < link_obj->u.object.len; i++) {
+        curr_key = link_obj->u.object.keys[i];
+
+        if (!strcmp(curr_key, "title"))
+            if (NULL == (parsed_link_name = YAJL_GET_STRING(link_obj->u.object.values[i])))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to get link name");
+    }
+
+    if (NULL == parsed_link_name)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "server response didn't contain link name");
+
+    if (NULL == (parsed_link_buffer = RV_malloc(strlen(parsed_link_name) + 1)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to allocate memory for link name");
+
+    memcpy(parsed_link_buffer, parsed_link_name, strlen(parsed_link_name) + 1);
+
+    *link_name = parsed_link_buffer;
+
+done:
+    if (parse_tree)
+        yajl_tree_free(parse_tree);
+
+    if (ret_value < 0) {
+        RV_free(*link_name);
+        *link_name = NULL;
+    }
+
+    return ret_value;
+} /* end RV_copy_link_name_by_index() */
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_copy_link_URI_by_index
+ *
+ * Purpose:     This callback is used to copy the URI of an link
+ *              in the server's response by index, to a provided buffer.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              May, 2023
+ */
+herr_t
+RV_copy_link_URI_by_index(char *HTTP_response, void *callback_data_in, void *callback_data_out)
+{
+    yajl_val           parse_tree = NULL, key_obj = NULL, link_obj = NULL;
+    const char        *parsed_link_URI = NULL;
+    H5VL_loc_by_idx_t *idx_params      = (H5VL_loc_by_idx_t *)callback_data_in;
+    hsize_t            index           = idx_params->n;
+    char             **link_URI        = (char **)callback_data_out;
+    const char        *curr_key        = NULL;
+    herr_t             ret_value       = SUCCEED;
+
```

Comment:
Same comment about checking `idx_params` as non-NULL here and assigning `index` after we're sure `idx_params` is non-NULL